Lib/unittest/loader.py

```diff
@@ -14,11 +14,6 @@
 
 __unittest = True
 
-# what about .pyc (etc)
-# we would need to avoid loading the same tests multiple times
-# from '.py', *and* '.pyc'
-VALID_MODULE_NAME = re.compile(r'[_a-z]\w*\.py$', re.IGNORECASE)
```

Comment:
I think it would be better to keep this global and update it to use `r'[^\W\d]\w*\.py$'`, but as @serhiy-storchaka pointed out in https://github.com/python/cpython/issues/68451#issuecomment-1093683815 this doesn't match all identifiers.

Perhaps a `is_valid_module_name()` global function could be added instead -- this will still provide a way to control what is valid, instead of hardcoding the check in `_find_test_path()`.

Reply:
I've added this as a private function in loader: `_is_valid_module_filename()`.  The code that makes use of the function makes use of it to tell if a filename is a valid python module name.  A different code path operates on directory names.  I think I put the amount of logic into the function that you were thinking (the equivalent of what the regex replaced).  Let me know if you were thinking of something with a larger scope.